src/pip/_internal/utils/temp_dir.py

```diff
@@ -193,10 +195,18 @@ def cleanup(self):
         """Remove the temporary directory created and reset state
         """
         self._deleted = True
-        if os.path.exists(self._path):
-            # Make sure to pass unicode on Python 2 to make the contents also
-            # use unicode, ensuring non-ASCII names and can be represented.
-            rmtree(ensure_text(self._path))
+        if not os.path.exists(self._path):
+            return
+        # Make sure to pass unicode on Python 2 to make the contents also
+        # use unicode, ensuring non-ASCII names and can be represented.
+        # This is only done on Windows because POSIX platforms use bytes
+        # natively for paths, and the bytes-text conversion omission avoids
+        # errors caused by the environment configuring encodings incorrectly.
+        if WINDOWS:
+            encoding = sys.getfilesystemencoding()
+            rmtree(ensure_text(self._path, encoding=encoding))
```

Comment:
Why do we have an explicit encoding here when previously we didn't? IMO, we should limit this to the absolute bare minimum change, as we won't ever have a chance to fix any bugs we might introduce (after 20.3, we'll be dropping Python 2 support, and we don't want to introduce new Python 2 bugs just before doing so).

Reply:
`ensure_text` uses `utf-8` by default, which is wrong on Windows in most of the cases. But fair enough—nobody has ever reported this, Python 2 is going away, and has enough problems on non-English Windows this probably does not matter.